packages/malloy/src/lang/ast/ast-main.ts

```diff
@@ -950,6 +1027,38 @@ export class FieldName
   get refString(): string {
     return this.name;
   }
+
+  get head(): FieldName {
+    return this;
+  }
+
+  get rest(): FieldName | undefined {
+    return undefined;
+  }
+}
+
+export class FieldPath extends FieldName implements FieldReferenceInterface {
```

Comment:
Maybe a path is a `ListOf<FieldName>` that also implements FieldReferenceInterface?

Reply:
I admit this is kinda weird.... but currently a `FieldPath` implements `FieldName`, so the `rest` could be a `FieldPath`. So the structure of `explore.join.join.join.fieldname` would be
```ts
new FieldPath(
  new FieldName("explore"),
  new FieldPath(
    new FieldName("join"),
    new FieldPath(
      new FieldName("join"),
      new FieldPath(
        new FieldName("join"),
        new FieldName("fieldname")
      )
    )
  )
)
```
But I'm going to try `Listof<FieldName>` because that makes more sense....